model/map/src/main/java/org/keycloak/models/map/common/locking/ConcurrentMapLocksUtils.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.common.locking;
```

Comment:
I wonder why this class in in the module model-map, as it is used only from HotRod. What is the reason for this?

Reply:
I generalized the util class to be usable with other stores that implement `ConcurrentMap`. This may also be used with CHM, or file store.

Reply:
@ahus1 We discussed this with Hynek and we decided this probably won't be useful for other storage, so I moved back to hotrod module.